driver/cabal-doctest.hs

```diff
@@ -0,0 +1,36 @@
+{-# LANGUAGE LambdaCase #-}
+module Main (main) where
+
+import           Prelude
+
+import           Data.Version
+import           System.Environment
+import           System.Exit
+import           System.FilePath
+import           System.IO.Temp (withSystemTempDirectory)
+import           System.Process
+
+import           Paths_doctest (version)
+
+main :: IO ()
+main = do
+  args <- getArgs
+  lookupEnv "CABAL" >>= \ case
+    Nothing -> run "cabal" args
+    Just cabal -> run cabal (drop 1 args)
+
+run :: String -> [String] -> IO ()
+run cabal args = withSystemTempDirectory "doctest" $ \ dir -> do
+  let
+    doctest = dir </> "doctest"
+    script = dir </> "init-ghci"
+  callProcess cabal ["install", "--ignore-project", "--installdir", dir, "--install-method=symlink", "doctest-" ++ showVersion version]
+  callProcess (dir </> "doctest") ["--version"]
+  callProcess cabal ("build" : "--only-dependencies" : args)
+  writeFile script ":seti -w -Wdefault"
```

Comment:
It would be nice if we could set those options via command-line flags, but I am not aware of a way to do that.

@mpickering @wismill do you have any input?